packages/valory/skills/transaction_settlement_abci/behaviours.py

```diff
@@ -533,23 +530,17 @@ def _check_tx_history(
         self,
     ) -> Generator[None, None, Tuple[VerificationStatus, Optional[str]]]:
         """Check the transaction history."""
-        history = (
-            self.synchronized_data.tx_hashes_history
-            if self.behaviour_id == CHECK_TX_HISTORY
-            else self.synchronized_data.late_arriving_tx_hashes
-        )
-
-        if not history:
+        if not self.history:
```

Comment:
Refactor to avoid conditioning on id (which would require unnecessary hardcoding)